list/list.js

```diff
@@ -273,6 +280,112 @@ var DefineList = Construct.extend("DefineList",
 			}
 			return this;
 		},
+		/**
+		 * @function can-define/list/list.prototype.assign assign
+		 * @parent can-define/list/list.prototype
+		 *
+		 * Sets items or properties on a list.
+		 *
+		 * @signature `list.assign(newProps)`
+		 *
+		 * Assigns the properties on the list with `newProps`. Properties not present in `newProps` will be left unchanged.
+		 *
+		 * ```js
+		 * var list = new DefineList(["A","B"]);
+		 * list.assign({count: 1000, skip: 2});
+		 * list.get("count") //-> 1000
+		 * ```
+		 */
+		assign: function(prop) {
+			if (canReflect.isListLike(prop)) {
```

Comment:
does `canReflect.assign` do the right thing here w/o you needing to feature test?